src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts

```diff
@@ -0,0 +1,105 @@
+import { BN } from '@polkadot/util';
+import { decodeAddress } from '@polkadot/util-crypto';
+import { XcmTokenInformation } from 'src/modules/xcm';
+import { container } from 'src/v2/common';
+import { ExtrinsicPayload, IApi, IApiFactory } from 'src/v2/integration';
+import { Asset } from 'src/v2/models';
+import { XcmChain } from 'src/v2/models/XcmModels';
+import { TokensAccounts } from 'src/v2/repositories/implementations/xcm/AcalaXcmRepository';
+import { Symbols } from 'src/v2/symbols';
+import { XcmRepository } from '../XcmRepository';
+
+/**
+ * Used to transfer assets from Bifrost
+ */
+export class BifrostXcmRepository extends XcmRepository {
+  constructor() {
+    const defaultApi = container.get<IApi>(Symbols.DefaultApi);
+    const apiFactory = container.get<IApiFactory>(Symbols.ApiFactory);
+    const registeredTokens = container.get<XcmTokenInformation[]>(Symbols.RegisteredTokens);
+    super(defaultApi, apiFactory, registeredTokens);
+    this.astarTokens = {
+      ASTR: 3,
+    };
+  }
+
+  public async getTransferCall(
+    from: XcmChain,
+    to: XcmChain,
+    recipientAddress: string,
+    token: Asset,
+    amount: BN
+  ): Promise<ExtrinsicPayload> {
+    if (!to.parachainId) {
+      throw `Parachain id for ${to.name} is not defined`;
+    }
+    let tokenData;
+    if (token.originAssetId == 'BNC') {
+      tokenData = { Native: 'BNC' };
+    } else if (token.originAssetId == '0') {
+      tokenData = { VToken2: 0 };
+    } else if (token.originAssetId == 'ASTR') {
+      tokenData = { Token2: 3 };
+    }
+    const destination = {
+      V1: {
+        parents: '1',
+        interior: {
+          X2: [
+            {
+              Parachain: to.parachainId,
+            },
+            {
+              AccountId32: {
+                network: {
+                  Any: null,
+                },
+                id: decodeAddress(recipientAddress),
+              },
+            },
+          ],
+        },
+      },
+    };
+    const destWeight = {
+      limited: new BN(10).pow(new BN(9)).muln(5),
+    };
+    return await this.buildTxCall(
+      from,
+      'xTokens',
+      'transfer',
+      tokenData,
+      amount,
+      destination,
+      destWeight
+    );
+  }
+
+  public async getTokenBalance(
+    address: string,
+    chain: XcmChain,
+    token: Asset,
+    isNativeToken: boolean
+  ): Promise<string> {
+    const api = await this.apiFactory.get(chain.endpoint);
+    try {
+      if (token.originAssetId == 'BNC') {
+        console.log(isNativeToken);
+        return (await this.getNativeBalance(address, chain)).toString();
+      } else if (token.originAssetId == '0') {
```

Comment:
what is '0'? Please make a const or add some comments.

Reply:
vToken2: 0 represents vDOT on our chain

Reply:
Please to use constant here to define vDOT. Do the same for ASTR value on your chain